File: solutions/csharp/protein-translation/1/ProteinTranslation.cs
using System;
using System.Collections.Generic;

public static class ProteinTranslation
{
    private static string ConvertToProtein(string input)
    {
        switch (input)
        {
            case "AUG":
                return "Methionine";
            case "UGG":
                return "Tryptophan";
            case "UGU":
            case "UGC":
                return "Cysteine";
            case "UUA":
            case "UUG":
                return "Leucine";
            case "UUU":
            case "UUC":
                return "Phenylalanine";
            case "UAU":
            case "UAC":
                return "Tyrosine";
            case "UAA":
            case "UAG":
            case "UGA":
                return "STOP";
            case "UCU":
            case "UCC":
            case "UCA":
            case "UCG":
                return "Serine";
            default:
                throw new Exception("Invalid sequence");
        }
    }

    public static string[] Proteins(string strand)
    {
        var results = new List<string>();
        for (int index = 0; index < strand.Length / 3; index++)
        {
            var protein = ConvertToProtein(strand.Substring(3 * index, 3));
            if (protein == "STOP") break;
            results.Add(protein);
        }
        return results.ToArray();
    }
}

File: solutions/python/protein-translation/1/protein_translation.py
translation_table = {
    "AUG": "Methionine",
    "UUU": "Phenylalanine",
    "UUC": "Phenylalanine",
    "UUA": "Leucine",
    "UUG": "Leucine",
    "UCU": "Serine",
    "UCC": "Serine",
    "UCA": "Serine",
    "UCG": "Serine",
    "UAU": "Tyrosine",
    "UAC": "Tyrosine",
    "UGU": "Cysteine",
    "UGC": "Cysteine",
    "UGG": "Tryptophan",
    "UAA": None,
    "UAG": None,
    "UGA": None,
}


def proteins(strand):
    proteins_list = [
        translation_table[strand[index : index + 3]]
        for index in range(0, len(strand), 3)
    ]
    if None in proteins_list:
        return proteins_list[0 : proteins_list.index(None)]
    return proteins_list
